src/Ssh/Ssh/SshCommands/EnterAzVMCommand.cs

```diff
@@ -142,6 +137,77 @@ private int StartSSHConnection()
             return sshProcess.ExitCode;
         }
 
+        private int StartRDPConnection()
+        {
+            Process sshProcess = null;
+            int success = 1;
+
+            try
+            {
+                // Get an open local port to act an a listener
+                TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
+                listener.Start();
+                rdpLocalPort = ((IPEndPoint)listener.LocalEndpoint).Port;
+                listener.Stop();
+
+                // Start SSH Process
+                sshProcess = CreateSSHProcess();
+                sshProcess.StartInfo.RedirectStandardError = true;
+                sshProcess.Start();
+
+                bool sshSuccess = WaitSSHTunnelConnection(sshProcess);
+                DoCleanup();
+
+                if (sshSuccess && !sshProcess.HasExited)
+                {
+                    string rdpCommand = GetClientApplicationPath("mstsc");
+
+                    Process rdpProcess = new Process();
+                    rdpProcess.StartInfo.FileName = rdpCommand;
+                    rdpProcess.StartInfo.Arguments = $"/v:localhost:{rdpLocalPort}";
+                    rdpProcess.Start();
+                    rdpProcess.WaitForExit();
+                    success = rdpProcess.ExitCode;
+                }
+            }
+            finally
+            {
+                if (sshProcess != null && !sshProcess.HasExited)
+                {
+                    sshProcess.Kill();
+                    sshProcess.WaitForExit();
+
+                    if (SshLogsPrinted() || IsDebugMode())
+                    {
+                        var stderr = sshProcess.StandardError;
```

Comment:
Is stderr still valid after the process exits?  Can `stderr.Readline()` throw?

Reply:
Yes, stderr is still valid once the process exits. You can still read any messages that were sent to stderr in between the last time you read stderr and the moment the process exits.

Reply:
I don't think stderr.Readline() can throw as long as stderr is valid. I can add a check to make sure stderr is not null.

Reply:
I believe `stderr` is a `StreamReader`, in which case it can throw a `IOException` exception.  Not sure if you want to catch it or let it propagate.